File: regexp/re_1to99.pl
#!/usr/bin/perl

#
# reference from http://plog.longwin.com.tw/my_note-programming/2009/05/21/regex-check-range-1-99-2009
#
# an regular expression to match 1-99 without leading 0
#

use strict;
use warnings;

my $rep = qr(^([1-9][0-9]?|)$);
my @test = qw(1 01 2 03 5 8 13 21 34 101);	# 01, 03, 101 not matched

foreach (@test)  {
	print $_, "\t";
	if ( m/$rep/ )  {
		print "matched and $1\n";
	}
	else  {
		print "not matched\n";
	}
}

File: regexp/re_1to99.py
#!/usr/bin/env python3

''' trivia regexp demo of python '''

import re

def main():
    ''' main '''
    test = ['1', '01', '2', '03', '5', '8', '13', '21', '34', '101']
    pattern = r"^([1-9][0-9]?|)$"
    p = re.compile(pattern)

    for i in test:
        m = p.match(i)
        print(i, "\t", end='')
        if m is None:
            print("is not matched")
        else:
            print("is matched: ", m.group(1))

if __name__ == '__main__':
    main()
